### PR TITLE
[MPS] Enable forward test for renorm

### PR DESCRIPTION
**Pull request by arunppsg**
Enabled forward test for renorm

cc @qqaatw

### CODE REVIEW

**Comment by qqaatw** [comment]
Hi, can you fix the tests, please?

**Comment by arunppsg** [comment]
Hey, I am not sure what is causing the issue. Can you please guide me on how to fix it?

**Comment by qqaatw** [comment]
@arunppsg, please fix the lint, thanks.

Here are a couple of things:

1. Please do not merge your PR when the newly added op is not covered by tests, doing so may cause the PR to be reverted as the implementation may be incorrect. If you're not sure which tests should be enabled or xfailed, you could ask the maintainers for guidance.
2. The `maxnorm` input is a `Scalar`, which is actually a class structure. We should pass the underlying value of it, which is a C++ primitive data type, into the Metal kernel instead of passing the pointer of the `Scalar`. Additionally, Metal doesn't support `double` data type, so we should convert it to `float` in advance.
3. The backward of `renorm` has actually been implemented for all backends, see [here](https://github.com/pytorch/pytorch/blob/387f1ab104748e8d061d0b9ec3a80fd6d31e7537/tools/autograd/derivatives.yaml#L1363-L1365). The issue was with data type.

**Review by kulinseth — APPROVED** [review]

File: torch/csrc/autograd/FunctionsManual.cpp
Comment (albanD) [inline torch/csrc/autograd/FunctionsManual.cpp:1567]: I wonder if we should move toAccumulateType() to have a `is_gpu` (or device) flag to be able to treat the accumulation type the same way on cuda and mps.

**Review by albanD — APPROVED** [review]
Random rant about our code being too cuda specific, shouldn't block this PR!

File: torch/csrc/autograd/FunctionsManual.cpp
Comment (qqaatw) [reply torch/csrc/autograd/FunctionsManual.cpp:1567]: yeah I thought about this but was not sure whether the cuda accumulation types are applicable to mps or not. So didn't change this part.

Will investigate it.

**Pull request closed without merging**
